src/task-4.js

```diff
@@ -1,9 +1,56 @@
 
 export default function runRover(commandsList) {
-  // Change me!
+  const ship = {
+    directions: ['top', 'left', 'bottom', 'right'],
+    indexOfCurrDirection: 0,
+    x: 0,
+    y: 0,
+    go(distance) {
+      switch (this.directions[this.indexOfCurrDirection]) {
+        case 'top': this.y += distance;
+          break;
+        case 'left': this.x -= distance;
+          break;
+        case 'bottom': this.y -= distance;
+          break;
+        case 'right': this.x += distance;
+          break;
+        default: break;
+      }
+    },
+    turn(direction) {
+      if (direction === 'left') {
+        this.indexOfCurrDirection++;
+        if (this.indexOfCurrDirection === 4) { this.indexOfCurrDirection = 0; }
+      } else {
+        this.indexOfCurrDirection--;
+        if (this.indexOfCurrDirection === -1) { this.indexOfCurrDirection = 3; }
+      }
+    }
+  };
+  const commandsInCorrectOrder = [];
+  let minValue = Infinity;
+  commandsList.forEach(currentCommand => {
```

Comment:
Can we refactor this task? The purpose of this practice is to teach you working with specific array/object methods. Please do not use `forEach`. You can use `sort` to get command in the correct order.